server/pulp/server/util.py

```diff
@@ -16,19 +17,31 @@
 _logger = logging.getLogger(__name__)
 
 
+# support md5 for fips
+def md5():
+    def modified_md5(*args, **kwargs):
+        kwargs['usedforsecurity'] = False
+        return hashlib.md5(*args, **kwargs)
+
+    if 'usedforsecurity' in inspect.getargspec(hashlib.new)[0]:
+        return modified_md5
+    else:
+        return hashlib.md5
+
+
 # Number of bytes to read into RAM at a time when validating the checksum
 CHECKSUM_CHUNK_SIZE = 8 * 1024 * 1024
 
 # Constants to pass in as the checksum type in verify_checksum
-TYPE_MD5 = hashlib.md5().name
+TYPE_MD5 = md5()().name
 TYPE_SHA = 'sha'
 TYPE_SHA1 = hashlib.sha1().name
 TYPE_SHA256 = hashlib.sha256().name
 
 HASHLIB_ALGORITHMS = (TYPE_MD5, TYPE_SHA, TYPE_SHA1, TYPE_SHA256)
 
 CHECKSUM_FUNCTIONS = {
-    TYPE_MD5: hashlib.md5,
+    TYPE_MD5: md5,
```

Comment:
I think either this should be `md5()` or `def md5` should return the result when called and not a function. WDYT?

Reply:
Fixed.